### ISSUE TITLE
Enhancement: show errors if there are issues with NeuroML

### ISSUE THREAD

**Issue by JustasB** [openworm/org.geppetto#468]
There are cases where Geppetto just shows the loading spinner without giving an error or a message. 

There is a particular example of this happening when the synapse used in the NeuroML-defined network was not valid. Eventually, after a few posts involving the team it was discovered that this was the case. 

It would likely save time for the team in the future if an error message regarding the issue with supplied NeuroML could be shown to the user. This way, future users would be able to solve their own problems without involving the team.

E.g.: https://github.com/JustasB/Mig3DTest/blob/be8239f81adff74726231441385317344ba087cb/NeuroML2/Networks/Bulb_2MC_20GC.net.nml#L152

This happened in the course of working on a model here: https://github.com/OpenSourceBrain/MiglioreEtAl14_OlfactoryBulb3D/issues/16

**Comment by tarelli** [comment]
@JustasB yo :) This is very possible in principle, we have all the communication channels already in place for the NeuroML interpreter to raise an exception which is then displayed in the UI. @adrianq any reason why this can't happen?

**Comment by adrianq** [comment]
@tarelli @JustasB I talked about this to @pgleeson last week. We can (and should) raise an exception so that instead of the load spinner the user show an error message. However, if we do this at the moment the error won't be helpful as it would be a Geppetto specific error. What @JustasB is asking for is an error that let him know what is the problem with the NeuroML Model. For instance, in that issue he mentioned the error was: 'you are using a component type instead of a component and this is not neuroml compliant'.  @pgleeson suggested to call a method that we can find in org.neuroml.model (something like validate) whenever we have an exception loading the model and return the returned string as part of the error message. This method provides the kind of information Justas is looking for. I wanted to talk you before implementing it in order to check if you see any conceptual problem about it. I forgot to do it in our last meeting though. Too many things on my plate :(

**Comment by tarelli** [comment]
@adrianq I don't see a conceptual problem putting inside the exception a detail of the error that is domain specific. This is an enhancement for the near future, nothing to worry about now, I just wanted to know if in principle you see any problems with it :)

**Comment by JustasB** [comment]
Sounds good, guys. Thank you.

**Comment by adrianq** [comment]
@JustasB I have implemented what we discussed in this issue. If there is an error parsing and resolving the model we will try to get an explanation from the neuromlvalidator. If  we manage to get a message, it will be displayed in the browser. Otherwise we will show the exception... for sure it is going to be better than the loading spinner. See what you will see in the example you mentioned in the image attached. This will be part of the next release (in 2-3 weeks it will be released)

![screenshot from 2016-01-28 18 08 19](https://cloud.githubusercontent.com/assets/6850223/12654306/3634af66-c5eb-11e5-9395-a50ac939fc1a.png)
